iree/compiler/Conversion/HLOToLinalg/HLOToLinalgOnTensors.cpp

```diff
@@ -182,6 +182,120 @@ struct ConcatenateOpConversion
 };
 }  // namespace
 
+//===----------------------------------------------------------------------===//
+// mhlo.fft conversion patterns.
+//===----------------------------------------------------------------------===//
+
+namespace {
+
+/// Creats coefficients based on DFT definition, see
+/// https://en.wikipedia.org/wiki/Discrete_Fourier_transform
+Value getDFTMatmulCoeff(OpBuilder b, Location loc, RankedTensorType matrixType,
+                        bool isRealPart) {
+  int rank = matrixType.getRank();
+  auto initTensor = b.create<linalg::InitTensorOp>(
+      loc, /*dyn_size=*/ValueRange{}, matrixType.getShape(),
+      matrixType.getElementType());
+  SmallVector<StringRef, 2> loops(rank, getParallelIteratorTypeName());
+  // scale = 2 * pi / N
+  double kScale = 2 * acos(-1) / matrixType.getDimSize(0);
+  if (!isRealPart) kScale = -kScale;  // -sin(x) = sin(-x)
+  return b
```

Comment:
You can instead turn this into `const dense<[DFT_VALUES]>` and generate the values here. matrixType is already ranked and all you need to get `DFT_VALUES` is the rank.

Reply:
Good point! I was trying to handle dynamic shapes as well, but it's actually not supported and we're not close to that case.  Let's generate the matrix here, it also addresses accuracy issues I hit.